api/src/test/java/org/openmrs/api/db/hibernate/ProviderDAOTest.java

```diff
@@ -98,10 +101,45 @@ public void getProviderRole_shouldReturnNullIfNotExists() {
 	public void getProviderRoleByUuid_shouldReturnTheProviderRoleIfExists() {
 		assertNotNull(providerDao.getProviderRoleByUuid("db7f523f-27ce-4bb2-86d6-6d1d05312bd5"));
 	}
-	
+
+	/**
+	 * @see ProviderDAO#getProviderRoleByUuid(String)
+	 */
 	@Test
 	public void getProviderRoleByUuid_shouldReturnNullIfNotExists() {
 		assertNull(providerDao.getProviderRoleByUuid("wrong-uuid"));
 	}
+
+	/**
+	 * @see ProviderDAO#getProvidersByRoles(List, boolean) 
+	 */
+	@Test
+	public void getProvidersByRoles_shouldGetProvidersByRoleIfRetiredFalse() {
+		Provider providerToRetire = Context.getProviderService().getProvider(1);
+		Context.getProviderService().retireProvider(providerToRetire, "test");
```

Comment:
Same as above. Is there a reason why we are duplicating this test?

Reply:
I still do not understand what this tests is trying to achieve. Can you explain it a bit more?

Reply:
This is to check whether the includeRetired is working or not.

Reply:
Why `shouldGetProvidersByRole` for both true and false? I find that confusing.

Reply:
> Why `shouldGetProvidersByRole` for both true and false? I find that confusing

This is just testing one instance where we want to fetch providers by Role that are not retired, and another instance where we want to fetch both retired and non-retired providers by their role.

Reply:
Can the name be made clearer to reflect that?